app/src/main/java/com/sdp/movemeet/utility/UserDatabase.java

```diff
@@ -0,0 +1,92 @@
+package com.sdp.movemeet.utility;
+
+import android.os.Build;
+
+import androidx.annotation.RequiresApi;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InterruptedIOException;
+import java.nio.file.Files;
+import java.util.HashMap;
+import java.util.Map;
+
+public class UserDatabase {
+    private final String filename = "users.json";
+    private Map<String, String> users = new HashMap<String, String>();
+
+    @RequiresApi(api = Build.VERSION_CODES.O)
+    public UserDatabase() throws IOException {
+        if (Files.exists(new File(filename).toPath())) {
+            readFromFile();
+        } else {
+            Files.createFile(new File(filename).toPath());
+            /*
+            try {
+                Files.createFile(new File(filename).toPath());
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+             */
+        }
+    }
+
+    public void readFromFile() throws IOException {
+        ObjectMapper mapper = new ObjectMapper();
+        users = mapper.readValue(new File(filename), Map.class);
+        /*
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            users = mapper.readValue(new File(filename), Map.class);
+        } catch (FileNotFoundException e) {
+            System.out.println("File not found. Change filename.");
+            e.printStackTrace();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+         */
+    }
+
+    public void writeToFile() throws IOException {
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.writeValue(new File(filename), users);
```

Comment:
Isn't it possible to simply put the new user added, instead of having to write the whole map and clean everytime ?

Reply:
That's correct, thanks for the idea. I'll think about optimizing this.

For most use-cases, there is usually 1-2 users on one device, so for the time being, we won't gain much from optimizing this. However, it would be nice to improve this in the future.